src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/SemanticTokensCapability.cs

```diff
@@ -1,11 +1,15 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.Collections.Generic;
 using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic
 {
     public class SemanticTokensCapability : DynamicCapability
     {
+        public IReadOnlyDictionary<string, uint> TokenTypes => SemanticTokenLegend.TokenTypesLegend;
```

Comment:
Does the LSP platform/client typically broadcast their "known" token types and modifiers?

Reply:
The client doesn't "broadcast" that, there's a set of "known" tokens mentioned a few times in the LSP spec that you can be pretty sure things will know how to interpret, but there's no process by which the client of some random IDE (say VIM) would say "If you return me a 'razorLanguageTagHelperStartTag' I'll be able to apply colors based on that.". Not 100% if that answers your question?

Reply:
But aren't capabilities typically from the client? Aka doesn't this represent what the client can potentially do? I'm confused why we're setting capabilities to our own